server.js

```diff
@@ -27,9 +51,99 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
+const queryBuilder = (req, res) => {
+  const {
+    /* destructure the query params */
+    start,
+    end,
+    'average-price-min': averagePriceMin,
+    'average-price-max': averagePriceMax,
+    'total-volume-min': totalVolumeMin,
+    'total-volume-max': totalVolumeMax,
+    region
+  } = req.query
+  /* Error handling */
+  if (start === undefined || end === undefined) {
+    /* Ensure we get dates, send 400 error otherwise */
+    return res.status(400).json({
+      status: 'failure',
+      message: 'Please ensure you pick two dates'
+    })
+  }
+  /* Add mandatory fields to the query */
+  const dbQuery = {
+    date: {
+      /* check for date range greater or equal && lesser than */
+      $gte: new Date(start),
+      $lt: new Date(end)
+    }
+  }
+  /* Adds optional fields */
+  if (averagePriceMin || averagePriceMax) {
+    dbQuery['averagePrice'] = {}
+    if (averagePriceMin) {
+      dbQuery['averagePrice']['$gte'] = +averagePriceMin
+    }
+    if (averagePriceMax) {
+      dbQuery['averagePrice']['$lte'] = +averagePriceMax
+    }
+  }
+  if (totalVolumeMin || totalVolumeMax) {
+    dbQuery['totalVolume'] = {}
+    if (totalVolumeMin) {
+      dbQuery['totalVolume']['$gte'] = +totalVolumeMin
+    }
+    if (totalVolumeMax) {
+      dbQuery['totalVolume']['$lte'] = +totalVolumeMax
+    }
+  }
+  if (region) {
+    dbQuery['region'] = { $eq: region }
+  }
+
+  console.log(dbQuery)
+  return dbQuery
+}
+
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/avocado-sales', async (req, res) => {
+  let page = req.query.page || 1
+
+  const dbQuery = queryBuilder(req, res)
+
+  const numberOfPages = Math.ceil(
+    (await AvocadoSales.countDocuments(dbQuery)) / MAXPERPAGE
+  ) /* Not really needed to send for each new page request - could store in frontend and pass bool to not run this function */
+
+  if (numberOfPages != 0 && numberOfPages < page) {
+    /* If page provided is too high return last page */
+    page = numberOfPages
+  }
+  const response = await AvocadoSales.find(dbQuery)
+    .skip(MAXPERPAGE * (page - 1))
+    .limit(MAXPERPAGE)
+
+  res.json({
+    'number-of-pages': numberOfPages,
+    'current-page': +page /* not really necessary as the client should know this but could be good for confirmation... or something.. */,
+    data: response
+  })
+})
+
+app.get('/avocado-sales/:id', async (req, res) => {
+  
+  try {
+    const result = await AvocadoSales.findOne({ _id: req.params.id })
+    if (result === null) {
+      throw 'error'
+    }
```

Comment:
Usually if you `throw` inside of a `catch` you should provide a more descriptive message for your catch statement, and then use that information to populate your response.